consul/src/main/java/org/secondbase/consul/ConsulModule.java

```diff
@@ -0,0 +1,217 @@
+package org.secondbase.consul;
+
+import com.google.common.net.HostAndPort;
+import com.orbitz.consul.AgentClient;
+import com.orbitz.consul.Consul;
+import com.orbitz.consul.model.agent.Registration;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.secondbase.core.SecondBase;
+import org.secondbase.core.SecondBaseException;
+import org.secondbase.core.config.SecondBaseModule;
+
+/**
+ * Consul module for SecondBase. Handles registration of services in Consul.
+ */
+public final class ConsulModule implements SecondBaseModule {
+
+    private static final Logger LOG = Logger.getLogger(ConsulModule.class.getName());
+
+    private static final ScheduledExecutorService consulKeepAlive
+            = Executors.newSingleThreadScheduledExecutor();
+    private static final long keepAliveInitialDelaySec = 1L;
+    private static final long keepAlivePeriodSec = 30L;
+    private static final DeregisterThread deregisterThread = new DeregisterThread();
+    private static final AtomicBoolean isDeregisterThreadActive = new AtomicBoolean(false);
+
+    private static ConsulModule consulModule;
+    private static AgentClient agentClient;
+
+    /**
+     * Deregister a list of services. To be used for deregistering consul services when
+     * shutting down.
+     */
+    private static class DeregisterThread extends Thread {
```

Comment:
It's just functionality of the class, really. And Flags are shared between modules. But I can move it out to.